hypothesis-python/tests/cover/test_functions.py

```diff
@@ -125,3 +129,61 @@ def test_functions_strategy_with_kwonly_args(f):
         f(1, 2)
     f(1, kwonly_arg=2)
     f(kwonly_arg=2, arg=1)
+
+
+def pure_func(arg1, arg2):
+    ...
+
+
+@given(
+    f=functions(like=pure_func, returns=integers(), pure=True),
+    arg1=integers(),
+    arg2=integers()
+)
+def test_functions_pure_with_same_args(f, arg1, arg2):
+    assert f(arg1, arg2) == f(arg1, arg2)
+
+
+@given(
+    f=functions(like=pure_func, returns=integers(), pure=True),
+    arg1=integers(),
+    arg2=integers()
+)
+def test_functions_pure_with_different_args(f, arg1, arg2):
+    r1 = f(arg1, arg2)
+    r2 = f(arg2, arg1)
+    assume(r1 != r2)
+    assert r1 != r2
```

Comment:
```suggestion
    # If this is never true, the test will fail with Unsatisfiable
```

The `assert` statement is redundant here, since it can never fail (absent `__ne__` hacks).  If we never find an example such that `r1 != r2`, the test will raise `Unsatisfiable` for us.